addons/iot_drivers/tools/certificate.py

```diff
@@ -76,13 +76,13 @@ def download_odoo_certificate():
         response = requests.post(
             'https://www.odoo.com/odoo-enterprise/iot/x509',
             json={'params': {'db_uuid': db_uuid, 'enterprise_code': enterprise_code}},
-            timeout=10,
+            timeout=95,  # let's encrypt library timeout
         )
         response.raise_for_status()
         response_body = response.json()
     except (requests.exceptions.RequestException, ValueError):
         _logger.exception("An error occurred while trying to reach odoo.com")
-        return None
+        return download_odoo_certificate()  # retry, we SHOULD get a certificate
```

Comment:
it does not sound like a great idea to me to infinitely recurse on that
What if they block odoo.com in their firewall or it have no internet access ?